book/src/qobject/qobject_struct.md

```diff
@@ -5,43 +5,151 @@ SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+{{#include ../../../examples/qml_features/rust/src/custom_base_class.rs:book_base_include}}
+    }
+```
+
+[Full Example](https://github.com/KDAB/cxx-qt/blob/main/examples/qml_features/rust/src/custom_base_class.rs)
 
-The `#[cxx_qt::qobject]` marked struct allows you to define the following items
+## Properties
 
-  * The name of the C++ class for the QObject
-  * Invokable methods that are exposed to Qt
-  * Private methods and fields for struct to use (eg this is useful for storing the channels for [threading](../concepts/threading.md))
-  * Mutate C++ state with [`CppObj`](./cpp_object.md)
-  * Use [CxxQtThread<T>](./cxxqtthread.md) for multi threaded workflows
+Fields within the `#[cxx_qt::qobject]` marked struct can be tagged with `#[qproperty]` to be exposed as [`Q_PROPERTY`s](https://doc.qt.io/qt-6/properties.html) on the generated QObject:
 
 ```rust,ignore,noplayground
-{{#include ../../../examples/qml_features/rust/src/rust_obj_invokables.rs:book_macro_code}}
+{{#include ../../../examples/qml_features/rust/src/properties.rs:book_properties_struct}}
 ```
 
-## Invokables
+Any type that CXX supports may be marked as a `#[qproperty]`.
+See the [Types page](../concepts/types.md) for a list of supported types.
 
-A `impl qobject::T` is used to define invokables, the `impl qobject::T` defines that the methods are implemented onto the C++ QObject `T`.
-Therefore they have access to both C++ and Rust methods. Also CXX-Qt adds wrapper code around your invokables to automatically perform any conversion between the [C++ and Rust types](../concepts/types.md).
+For every `#[qproperty]`, CXX-Qt will generate setters and getters, as well as a "changed" signal.
 
-To mark a method as invokable simply add the `#[qinvokable]` attribute to the Rust method. This then causes `Q_INVOKABLE` to be set on the C++ definition of the method, allowing QML to call the invokable.
+On The C++ side:
+  * setter: `set<Property>`
+  * getter: `get<Property>`
 
-Note to access properties on the C++ object use [Cpp Object](./cpp_object.md).
+On the Rust side:
+  * setter: `set_<Property>`
+  * getter: `<Property>`
 
-## Properties
+where `<Property>` is the name of the property.
+
+These setters and getters assure that the changed signal is emitted every time the property is edited.
 
-Fields within the `#[cxx_qt::qobject]` marked struct can be tagged with the `#[qproperty]`, this declares which fields should be exposed as `Q_PROPERTY`s on the QObject.
+Any field that's not marked as `#[qproperty]` won't be accessible from C++, but it will be accessible from Rust.
+See the [Private fields section](#private-methods-and-fields)
+
+Then `Default` needs to be implemented for the struct either by hand or you can of course also use `#[derive(Default)]`.
 
 ```rust,ignore,noplayground
-{{#include ../../../examples/qml_features/rust/src/struct_properties.rs:book_macro_code}}
+{{#include ../../../examples/qml_features/rust/src/properties.rs:book_properties_default}}
+```
```

Comment:
This is out of place in the section about properties. How about moving this to a new section?

Reply:
Potentially, it's related to properties in that when you define a property you have to give it a default though, so the new section shouldn't be far away...

Reply:
Done